### PR TITLE
Add observer mode

### PR DESCRIPTION
**Pull request by daniel-k-taylor**

### CODE REVIEW

File: scenes/menu/main_menu.gd [hunk 1]
```diff
@@ -315,3 +374,25 @@ func _on_bgm_check_box_toggled(button_pressed : bool):
...
+	room_select.text = selected_match['name']
+	_on_join_button_pressed()
+
+func _on_modal_list_observe_match_pressed(row_index):
+	var matches = NetworkManager.get_match_list()
+	var selected_match = matches[row_index]
+	var room_name = selected_match['name']
+	var player_name = get_player_name()
+	NetworkManager.observe_room(player_name, room_name)
+	update_buttons(true)
```
Comment (Mach131) [inline scenes/menu/main_menu.gd:397]: Would it be too much extra effort to support people opting out of being observed? (e.g. players can press a checkbox before entering a room/matchmaking; if either player has it enabled then the match object stores a flag that this checks)

**Review by Mach131 — COMMENTED** [review]
Seems good at a high level; I just remember it being brought up that this being optional would likely be a good idea, if that's feasible

File: scenes/menu/main_menu.gd (hunk 1 above)
Comment (daniel-k-taylor) [reply scenes/menu/main_menu.gd:397]: do we really need that? I don't think I would allow it for the matchmaking queue. And nobody really even uses rooms.

Comment (Mach131) [reply scenes/menu/main_menu.gd:397]: I do see a fair number of rooms occasionally, but yee fair enough

**Review by Mach131 — APPROVED** [review]

**Pull request merged**